### ISSUE TITLE
[Chip] Ensure the `label` prop is provided to assistive technologies

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#8696]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

Chip's `label` property should be provided over the `value` or slotted text.

### Actual Behavior

The component's `label` property provides context to AT, where `value` is not provided to assistive technologies or when a parent `chip-group` is present.

### Expected Behavior

As part of an upcoming breaking change release, we should consider making `label` required from our discussions in https://github.com/Esri/calcite-design-system/pull/6075#discussion_r1055903130. This way the internals stay the same and label is read to assistive technologies, and required. Value should just be optionally used for getting value in selection mode.

If enforcing a required `label`, the `value` can be optional and not read back to assistive technologies. This seems to be a similar pattern in other components, such as `alert` and the parent `chip-group`:

https://github.com/Esri/calcite-design-system/blob/107792d52159848bd6e01fd14fa80292fbfd02e6/packages/calcite-components/src/components/alert/alert.tsx#L111-L112

https://github.com/Esri/calcite-design-system/blob/107792d52159848bd6e01fd14fa80292fbfd02e6/packages/calcite-components/src/components/chip-group/chip-group.tsx#L46-L47

 cc @macandcheese 

### Reproduction Sample

https://codepen.io/geospatialem/pen/poYKVYv

### Reproduction Steps

1. Enable JAWS
2. Open the [sample](https://codepen.io/geospatialem/pen/poYKVYv) in a Chrome browser
3. Observe the example for the `chip-group` works as-expected
    - "Light gray basemap" and "Dark gray basemap" contexts are provided
5. Using the arrow keys navigate to the second and first examples
6. Observe the `label`'s are not provided to AT
  - Expected "Gray basemap" for both examples

### Reproduction Version

`2.4.0`

### Relevant Info

- Windows
- JAWS
- Chrome browser

### Priority impact

p4 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by macandcheese** [comment]
In addition to Chip's `value` property becoming optional - I'd propose we also make the referenced Block `heading` property optional and add a required `label` property to that component. 

That would facilitate using Block as container without a displayed text heading, while providing AT with the `label`.

**Comment by geospatialem** [comment]
Additional effort is needed to ensure the `label` context is provided to AT, currently if the `chip` does not have slotted text, no context is provided Instead, it should have similar support to `button`:

- Testing codepen with `3.0.0-next.29`: https://codepen.io/geospatialem/pen/mybbzZP
- Still trying to get the hang of Lit/Lumina, but this is the `button` code:
https://github.com/Esri/calcite-design-system/blob/0362b4e661a2b009fd26c2eea1b25f572ce0b5ef/packages/calcite-components/src/components/button/button.tsx#L343


cc @benelan

**Comment by geospatialem** [comment]
Verified with JAWS and NVDA in `3.0.0-next.37`
